src/modules/components/proposal/proposalActions/actions/proposalActionTokenMint/proposalActionTokenMint.tsx

```diff
@@ -0,0 +1,82 @@
+import { DataList, DefinitionList, formatterUtils, Heading, NumberFormat } from '../../../../../../core';
+import { type IProposalActionTokenMintMetadataReceiver } from '../../../../../types';
+import { MemberDataListItemStructure } from '../../../../member';
+import { type IProposalActionTokenMint } from '../../proposalActionsTypes/proposalActionTokenMint';
+
+export interface IProposalActionTokenMintProps {
+    /**
+     *  Token mint action
+     */
+    action: IProposalActionTokenMint;
+}
+
+const calculateFormattedTokensMinted = (receivers: IProposalActionTokenMintMetadataReceiver[]) => {
+    return receivers.map((receiver) => {
+        const tokensMinted = receiver.newBalance - receiver.currentBalance;
+        return {
+            ...receiver,
+            tokensMinted,
+        };
+    });
+};
+
+export const ProposalActionTokenMint: React.FC<IProposalActionTokenMintProps> = (props) => {
+    const { action } = props;
+
+    const newTokens = action.receivers.reduce((sum, receiver) => {
+        /* For each receiver, calculate the difference between the new balance and the current balance. Keep track of the difference to get the total number of new tokens distributed */
+        return sum + (receiver.newBalance - receiver.currentBalance);
+    }, 0);
+
+    const newHolders = action.receivers.reduce((count, receiver) => {
+        /* Check if the receiver's current balance is 0, which means they are a new holder. If true, increment the count by 1. */
+        return count + (receiver.currentBalance === 0 ? 1 : 0);
+    }, 0);
+
+    // Formatted values
+    const formattedNewTokens = formatterUtils.formatNumber(newTokens, { format: NumberFormat.TOKEN_AMOUNT_SHORT });
+    const formattedNewHolders = formatterUtils.formatNumber(newHolders, { format: NumberFormat.GENERIC_SHORT });
+    const formattedTotalTokenSupply =
+        formatterUtils.formatNumber(action.tokenSupply, { format: NumberFormat.TOKEN_AMOUNT_SHORT }) ?? 0;
+    const formattedTotalTokenHolders =
+        formatterUtils.formatNumber(action.holdersCount, { format: NumberFormat.GENERIC_SHORT }) ?? 0;
+
+    const formattedReceivers = calculateFormattedTokensMinted(action.receivers);
+
+    return (
+        <div data-testid="proposal-action-token-mint-container" className="flex flex-col gap-8">
+            <DataList.Root entityLabel="proposalActionTokenMintReceivers">
+                <DataList.Container className="grid grid-cols-1 gap-2 md:grid-cols-2">
+                    {formattedReceivers.map((receiver) => (
+                        <MemberDataListItemStructure
+                            key={receiver.address}
+                            className="w-full"
+                            address={receiver.address}
+                            tokenAmount={receiver.tokensMinted}
```

Comment:
Given that we're only doing a simple operation on the `calculateFormattedTokensMinted` function, I would just make it on the render function, wdyt?
```suggestion
                    {receivers.map(({ address, newBalance, currentBalance }) => (
                        <MemberDataListItemStructure
                            key={address}
                            className="w-full"
                            address={address}
                            tokenAmount={newBalance - currentBalance}
```

Reply:
Yeah I agree with this, initially I had overcomplicated this logic, which is why I ended up moving it outside the JSX, but yeah this makes a lot more sense.